packages/react-core/src/components/Select/Select.tsx

```diff
@@ -481,13 +490,47 @@ export class Select extends React.Component<SelectProps & OUIAProps, SelectState
       inputIdPrefix,
       /* eslint-enable @typescript-eslint/no-unused-vars */
       menuAppendTo,
+      favorites,
+      onFavorite,
+      favoritesLabel,
       ...props
     } = this.props;
     const { openedOnEnter, typeaheadInputValue, typeaheadActiveChild, typeaheadFilteredChildren } = this.state;
     const selectToggleId = toggleId || `pf-select-toggle-id-${currentId++}`;
     const selections = Array.isArray(selectionsProp) ? selectionsProp : [selectionsProp];
     const hasAnySelections = Boolean(selections[0] && selections[0] !== '');
     let childPlaceholderText = null as string;
+
+    // If onFavorites is set,  add isFavorite prop to children and add a Favorites group to the SelectMenu
+    let renderableItems: React.ReactNode[] = [];
+    if (onFavorite) {
+      let favoritesGroup: React.ReactNode[] = [];
+      let renderableFavorites: React.ReactNode[] = [];
+      if (favorites.length > 0) {
+        // if variant is typeahead use filtered children
+        const tempRenderableChildren =
+          variant === 'typeahead' || 'typeaheadmulti' ? typeaheadFilteredChildren : children;
```

Comment:
```suggestion
          variant === 'typeahead' || variant === 'typeaheadmulti' ? typeaheadFilteredChildren : children;
```

Reply:
is there a test to check if this works properly with both variants?

Reply:
Yes.  in the demo app